### DISCUSSION TITLE
How to execute a command without getting it added to history, e.g. with passwords?

### DISCUSSION THREAD

**Discussion by fabiang** [nushell/nushell#8884] [Q&A] [answered]
Hello,

is there any way of doing this? In fish/bash conversion is to add a space before the command. Doing this in Nu doesn't work:

```
 let-env TEST = "foo"; nu -c "history | last 1"

╭──────┬─────────────────────────────────────────────────╮
│    # │                     command                     │
├──────┼─────────────────────────────────────────────────┤
│ 2576 │  let-env TEST = "foo"; nu -c "history | last 1" │
╰──────┴─────────────────────────────────────────────────╯
```

For whatever reason my bash wasn't configured with HISTCONTROL= environment variable, so I had to set it manually.
```
[fabian.grutschus@NBINV0026 ~]$ HISTCONTROL=ignoreboth
[fabian.grutschus@NBINV0026 ~]$ export TEST="foo";
[fabian.grutschus@NBINV0026 ~]$  export TEST="foo";
[fabian.grutschus@NBINV0026 ~]$ history | tail -n3
  528  HISTCONTROL=ignoreboth
  529  export TEST="foo";
  530  history | tail -n3

```

In fish:
```
~ $ echo "test"
test
~ $ set TEST 1
~ $  set TEST 2
~ $ history | head -n2
set TEST 1
echo "test"
```

**Comment by fdncred** [comment]
I believe there's a PR for this right now in the reedline repo waiting for approval.

**Comment by ghost** [comment]
a "trick" for the meanwhile, which i use: `let passwd = (input -s)` allows you to write the password to a variable without writing it to the history

**Comment by fabiang** [comment, marked as answer]
[Nushell 0.82](https://www.nushell.sh/blog/2023-06-27-nushell_0_82.html)

> REPL: Mark line entries as excluded from the history by leading with a space (samlich)
In some situations you may want to avoid storing an entry inside your shell history. For example if it includes a secret or running it again in a different context could be dangerous. To stop nushell from storing a particular line in its history you can now start with a space and it will be ignored. You can still recall your last entry once by pressing the Up-Arrow but the command will never be written to disk and disappear as soon as you submit another command.
